### PR TITLE
Improve docstring of Albert

### PR DESCRIPTION
**Pull request by soma2000-lang**
Similar to #843
@mattdangerw @jbischof @chenmoneygithub

### CODE REVIEW

File: keras_nlp/models/albert/albert_classifier.py [hunk 1]
```diff
@@ -107,19 +66,22 @@ class AlbertClassifier(Task):
         sequence_length=128,
     )
 
-    # Create a AlbertClassifier and fit your data.
+    #Pretrained classifier.
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_classifier.py:69]: Space after `#`

File: keras_nlp/models/albert/albert_classifier.py [hunk 2]
```diff
@@ -133,17 +95,42 @@ class AlbertClassifier(Task):
...
-        loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
-    )
     classifier.fit(x=preprocessed_features, y=labels, batch_size=2)
     ```
+
+     Custom backbone and vocabulary.
+    ```python
+    features = ["The quick brown fox jumped.", "I forgot my homework."]
+    labels = [0, 3]
+    vocab = ["[UNK]", "[CLS]", "[SEP]", "[PAD]", "[MASK]"]
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_classifier.py:111]: I'm pretty sure this will not work. You will need something like below as albert is based on sentencepiece. Please make sure to test out your code!

```python
bytes_io = io.BytesIO()
sentencepiece.SentencePieceTrainer.train(
    sentence_iterator=iter(["The quick brown fox jumped."]),
    model_writer=bytes_io,
    vocab_size=8,
    model_type="WORD",
    pad_id=0,
    unk_id=1,
    bos_id=2,
    eos_id=3,
    pad_piece="<pad>",
    unk_piece="<unk>",
    bos_piece="[CLS]",
    eos_piece="[SEP]",
    user_defined_symbols="[MASK]",
)
tokenizer = keras_nlp.tokenizers.AlbertTokenizer(proto=bytes_io.getvalue())
```

File: keras_nlp/models/albert/albert_classifier.py [hunk 3]
```diff
@@ -107,19 +66,22 @@ class AlbertClassifier(Task):
...
     )
 
-    # Create a AlbertClassifier and fit your data.
+    #Pretrained classifier.
     classifier = keras_nlp.models.AlbertClassifier.from_preset(
         "albert_base_en_uncased",
         num_classes=4,
         preprocessor=preprocessor,
     )
+    # Re-compile (e.g., with a new learning rate).
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_classifier.py:75]: I think you are missing in the bert classifier example where we run fit before  "re compiling"

File: keras_nlp/models/albert/albert_masked_lm_preprocessor.py [hunk 4]
```diff
@@ -68,6 +68,15 @@ class AlbertMaskedLMPreprocessor(AlbertPreprocessor):
                     left-to-right manner and fills up the buckets until we run
                     out of budget. It supports an arbitrary number of segments.
 
+        Call arguments:
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_masked_lm_preprocessor.py:71]: remove indent (should align with Args)

File: keras_nlp/models/albert/albert_masked_lm_preprocessor.py [hunk 5]
```diff
@@ -76,54 +85,37 @@ class AlbertMaskedLMPreprocessor(AlbertPreprocessor):
...
-    sentence = tf.constant("The quick brown fox jumped.")
-    preprocessor(sentence)
+    preprocessor("The quick brown fox jumped.")
 
     # Tokenize and mask a batch of sentences.
-    sentences = tf.constant(
-        ["The quick brown fox jumped.", "Call me Ishmael."]
-    )
-    preprocessor(sentences)
+    preprocessor("The quick brown fox jumped.", "Call me Ishmael.")
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_masked_lm_preprocessor.py:91]: inner args should be in a list

File: keras_nlp/models/albert/albert_masked_lm_preprocessor.py [hunk 6]
```diff
@@ -76,54 +85,37 @@ class AlbertMaskedLMPreprocessor(AlbertPreprocessor):
...
 
-    proto = bytes_io.getvalue()
-
-    tokenizer = keras_nlp.models.AlbertTokenizer(proto=proto)
-
-    preprocessor = keras_nlp.models.AlbertMaskedLMPreprocessor(
-        tokenizer=tokenizer
+    first = tf.constant(["The quick brown fox jumped.", "Call me Ishmael."])
+    second = tf.constant(["The fox tripped.", "Oh look, a whale."])
+    # Map single sentences.
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_masked_lm_preprocessor.py:108]: take a look at the source example, but you removed all the empty newlines, please add them back

File: keras_nlp/models/albert/albert_preprocessor.py [hunk 7]
```diff
@@ -33,30 +33,17 @@ class AlbertPreprocessor(Preprocessor):
 
     This preprocessing layer will do three things:
 
-     - Tokenize any number of input segments using the `tokenizer`.
-     - Pack the inputs together using a `keras_nlp.layers.MultiSegmentPacker`.
+     1.Tokenize any number of input segments using the `tokenizer`.
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_preprocessor.py:36]: space after period, or this wont render properly as a markdown list

File: keras_nlp/models/albert/albert_preprocessor.py [hunk 8]
```diff
@@ -70,71 +57,65 @@ class AlbertPreprocessor(Preprocessor):
                     "waterfall" algorithm that allocates quota in a
                     left-to-right manner and fills up the buckets until we run
                     out of budget. It supports an arbitrary number of segments.
+        Call arguments:
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_preprocessor.py:60]: Add newline, fix alignment

File: keras_nlp/models/albert/albert_preprocessor.py [hunk 9]
```diff
@@ -70,71 +57,65 @@ class AlbertPreprocessor(Preprocessor):
...
-    features = tf.constant(
-        ["The quick brown fox jumped.", "Call me Ishmael."]
+    # Custom vocabulary.
+    vocab = ["[UNK]", "[CLS]", "[SEP]", "[PAD]", "[MASK]"]
+    vocab += ["The", "quick", "brown", "fox", "jumped", "."]
+    tokenizer = keras_nlp.models.AlbertTokenizer(vocabulary=vocab)
+    preprocessor = keras_nlp.models.AlbertPreprocessor(tokenizer)
+    preprocessor("The quick brown fox jumped.")
+    ```
+    Mapping with `tf.data.Dataset`.
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_preprocessor.py:96]: Add empty newlines through this file to match the bert version

File: keras_nlp/models/albert/albert_tokenizer.py [hunk 10]
```diff
@@ -61,6 +61,13 @@ class AlbertTokenizer(SentencePieceTokenizer):
 
     # Detokenization.
     tokenizer.detokenize(tf.constant([[2, 14, 2231, 886, 2385, 3]]))
+
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_tokenizer.py:64]: ~10 lines up from here, you should be instantiating the tokenizer with `from_preset`

File: keras_nlp/models/albert/albert_tokenizer.py [hunk 11]
```diff
@@ -61,6 +61,13 @@ class AlbertTokenizer(SentencePieceTokenizer):
...
+    # Custom vocabulary.
+    vocab = ["[UNK]", "[CLS]", "[SEP]", "[PAD]", "[MASK]"]
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_tokenizer.py:66]: This will not work! As mentioned above you will need the sentencepiece trainer.

File: keras_nlp/models/albert/albert_masked_lm.py [hunk 12]
```diff
@@ -90,24 +95,9 @@ class AlbertMaskedLM(Task):
...
-    # Create a ALBERT masked LM and fit the data.
-    masked_lm = keras_nlp.models.AlbertMaskedLM(
-        backbone,
-        preprocessor=None,
-    )
-    masked_lm.compile(
-        loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
-        jit_compile=True
+    masked_lm = keras_nlp.models.AlbertMaskedLM.from_preset(
+    "albert_base_en_uncased",
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_masked_lm.py:99]: Fix indentation.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Left some comments.

Overall this needs two things.
 - Test out all the examples here, some are unrunnable. You can `pip install git+https://github.com/keras-team/keras-nlp.git --upgrade` in a colab to run our latest code changes.
 - Pay close attention to formatting and style. Make sure to indent and leave line breaks for clarify to match the original PR. Note that copying and pasting from github will UI remove all empty lines.

File: keras_nlp/models/albert/albert_backbone.py [hunk 13]
```diff
@@ -33,11 +33,12 @@ def albert_kernel_initializer(stddev=0.02):
 
 @keras_nlp_export("keras_nlp.models.AlbertBackbone")
 class AlbertBackbone(Backbone):
-    """ALBERT encoder network.
+    """An ALBERT encoder network.
 
     This class implements a bi-directional Transformer-based encoder as
     described in
-    ["ALBERT: A Lite BERT for Self-supervised Learning of Language Representations"](https://arxiv.org/abs/1909.11942).
+    ["ALBERT: A Lite BERT for Self-supervised Learning of Language Representations"]
```
Comment (chenmoneygithub) [inline keras_nlp/models/albert/albert_backbone.py:40]: Let's remove the linebreak here, it's okay to exceed the limit if it's a hyperlink.

File: keras_nlp/models/albert/albert_masked_lm_preprocessor.py (hunk 4 above)
Comment (chenmoneygithub) [reply keras_nlp/models/albert/albert_masked_lm_preprocessor.py:71]: This comment has not been addressed, please fix it, thanks!

File: keras_nlp/models/albert/albert_masked_lm_preprocessor.py (hunk 5 above)
Comment (chenmoneygithub) [reply keras_nlp/models/albert/albert_masked_lm_preprocessor.py:91]: this one too - ```preprocessor(["The quick brown fox jumped.", "Call me Ishmael."])```

File: keras_nlp/models/albert/albert_tokenizer.py (hunk 10 above)
Comment (chenmoneygithub) [reply keras_nlp/models/albert/albert_tokenizer.py:64]: the line is ```tokenizer = keras_nlp.models.AlbertTokenizer(proto="model.spm")```

**Review by chenmoneygithub — CHANGES_REQUESTED** [review]
Thanks for the PR! There seems to be some unaddressed comments, please fix them, thanks!

**Comment by mattdangerw** [comment]
This has landed elsewhere. (We are lining up these changes for an upcoming release)

Thanks for the contribution!
